gems/pending/appliance_console/date_time_configuration.rb

```diff
@@ -0,0 +1,96 @@
+require "appliance_console/logging"
+require 'appliance_console/prompts'
+
+module ApplianceConsole
+  class DateTimeConfiguration
+    DATE_REGEXP   = /^(2[0-9]{3})-(0?[1-9]|1[0-2])-(0?[1-9]|[12][0-9]|3[01])/
+    DATE_PROMPT   = "current date (YYYY-MM-DD)".freeze
+    TIME_REGEXP   = /^(0?[0-9]|1[0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9])/
+    TIME_PROMPT   = "current time in 24 hour format (HH:MM:SS)".freeze
+
+    attr_accessor :new_date, :new_time
+
+    include ApplianceConsole::Logging
+
+    def initialize(config = {})
+      @new_date = config[:new_date]
+      @new_time = config[:new_time]
+    end
+
+    def activate
+      say("Applying time configuration...")
+      disable_auto_sync &&
+        configure_date_time
```

Comment:
`configure_date_time if disable_auto_sync` returns `nil` when `disable_auto_sync` fails. The false is not bubble up.
Where the && chaining bubbles up the return `false`.
```ruby
def configure_date_time
  return true
end

def disable_auto_sync
  return false
end

configure_date_time if disable_auto_sync
=> nil

disable_auto_sync &&
  configure_date_time
=> false
```